src/cmdstan/command.hpp

```diff
@@ -90,19 +96,143 @@ std::shared_ptr<stan::io::var_context> get_var_context(const std::string file) {
+      }
+    };
+    // use default for all chain inits
+    if (file == "") {
+      using stan::io::dump;
+      std::fstream stream(file.c_str(), std::fstream::in);
+      return context_vector(n_chain, std::make_shared<dump>(dump(stream)));
+    } else {
+      size_t file_marker_pos = file.find_last_of(".");
+      if (file_marker_pos > file.size()) {
+        std::stringstream msg;
+        msg << "Found: \"" << file
+            << "\" but user specied files must end in .json or .csv";
+        throw std::invalid_argument(msg.str());
+      }
+      std::string file_name = file.substr(0, file_marker_pos);
+      std::string file_ending = file.substr(file_marker_pos, file.size());
+      if (file_ending != ".json" || file_ending != ".csv") {
+        std::stringstream msg;
+        msg << "file ending of " << file_ending
+            << " is not supported by cmdstan";
+        throw std::invalid_argument(msg.str());
+      }
+      std::string file_1
+          = std::string(file_name + "_" + std::to_string(1) + file_ending);
+      std::fstream stream_1(file_1.c_str(), std::fstream::in);
+      // Check if file_1 exists, if so then we'll assume n_chain of these exist.
+      if (stream_1.rdstate() & std::ifstream::failbit) {
+        // if that fails we will try to find a base file
+        std::fstream stream(file.c_str(), std::fstream::in);
+        if (stream.rdstate() & std::ifstream::failbit) {
+          std::string file_name_err = std::string(
+              "\"" + file_1 + "\" and base file \"" + file + "\"");
+          std::stringstream msg;
+          msg << "Searching for  \"" << file_name_err << std::endl;
+          msg << "Can't open either of specified files," << file_name_err
+              << std::endl;
+          throw std::invalid_argument(msg.str());
+        } else {
+          return context_vector(1, make_context(file, stream, file_ending));
+        }
+      } else {
+        // If we found file_1 then we'll assume file_{1...N} exists
+        context_vector ret;
+        ret.reserve(n_chain);
+        ret.push_back(make_context(file_1, stream_1, file_ending));
+        for (size_t i = 1; i < n_chain; ++i) {
+          std::string file_i
+              = std::string(file_name + "_" + std::to_string(i) + file_ending);
+          std::fstream stream_i(file_1.c_str(), std::fstream::in);
+          // If any stream fails at this point something went wrong with file
+          // names.
+          if (stream_i.rdstate() & std::ifstream::failbit) {
+            std::string file_name_err = std::string(
+                "\"" + file_1 + "\" but cannot open \"" + file_i + "\"");
+            std::stringstream msg;
+            msg << "Found " << file_name_err << std::endl;
+            throw std::invalid_argument(msg.str());
+          }
+          ret.push_back(make_context(file_i, stream_i, file_ending));
+        }
+        return ret;
+      }
+    }
+  }
+  // This should not happen
+  using stan::io::dump;
```

Comment:
should not happen => throw an error?

Reply:
So this is more of a weird C++ thing, we can't have the function reach the end without having a valid return. This code won't ever happen but this path does need to exist